utils/zerovec/src/varzerovec/owned.rs

```diff
@@ -0,0 +1,325 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use super::*;
+use core::fmt;
+use core::marker::PhantomData;
+use core::mem;
+use core::ptr;
+use core::slice;
+
+#[derive(Clone)]
+pub struct VarZeroVecOwned<T> {
+    marker: PhantomData<[T]>,
+    // safety invariant: must parse into a valid SliceComponents
+    entire_slice: Vec<u8>,
```

Comment:
I kinda thought it would be a duplication so I didn't. Bear in mind that reading the length _should_ be basically free since it's just a `mov` operation either way. Perhaps we can wait for Alan's benchmarks and play with this?